### ISSUE TITLE
concurrent ChoNACHAReader issue

### ISSUE THREAD

**Issue by ok1-ok1** [Cinchoo/ChoETL.NACHA#40]
I am not sure if it is related to ChoETL or ChoETL.NACHA, but I got some exceptions while running a couple of unittests:

Can't convert '121015' value from 'System.String' type to 'System.DateTime' type.
Can't convert '121015' value from 'System.String' type to 'System.DateTime' type.
Can't convert '1518' value from 'System.String' type to 'System.DateTime' type.

Below code can simulate the exceptions:
```
class Program
    {
        static void Main(string[] args)
        {
            for (int i = 0; i < 3; i++)
            {
                string filename = "20151027B0000327P018CHK_" + i + ".ACH";
                var t = Task.Run(() => Parse(filename));
            }

            Console.ReadLine();
        }

        public static List<object> Parse(string fileName)
        {
            System.Threading.Thread.Sleep(100);
            List<object> records = new List<object>();
            try
            {
                using (var file = new ChoNACHAReader(fileName))
                {
                    foreach (var record in file)
                    {
                        records.Add(record);
                    }
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }           

            return records;
        }
    } 
```

Thanks

**Comment by Cinchoo** [comment]
Pls specify the version of the library using? Your current culture info?

**Comment by ok1-ok1** [comment]
ChoETL.NACHA.NETStandard version 1.0.2
and 
Culture is set to en-US in app
```
System.Globalization.CultureInfo.DefaultThreadCurrentCulture = new System.Globalization.CultureInfo("en-US");
```

**Comment by ok1-ok1** [comment]
In addition, if I put a lock around the using block of above ChoNACHAReader, it works. What do you think?

**Comment by Cinchoo** [comment]
This is bug in ChoETL. Will be applying fix, Will let u know.

**Comment by Cinchoo** [comment]
Applied the fix to ChoETL v1.2.1.9. Pls update this package and try and let me know.

Thanks for reporting this issue.

**Comment by ok1-ok1** [comment]
Thanks, it works.
